src/actions/ui.js

```diff
@@ -61,15 +65,48 @@ export const acUpdateUiProgramStageId = (value, metadata) => ({
     metadata,
 })
 
+const tClearUiProgramDimensions = () => (dispatch, getState) => {
+    const { ui, metadata } = getState()
+    const idsToRemove = ui.layout.columns
+        .concat(ui.layout.filters)
+        .filter((dimensionId) =>
+            DIMENSION_TYPES_PROGRAM.has(metadata[dimensionId].dimensionType)
+        )
+    dispatch(acRemoveUiLayoutDimensions(idsToRemove))
+    dispatch(acRemoveUiItems(idsToRemove))
+}
+
+export const tClearProgramStageDimensions =
+    (stageId) => (dispatch, getState) => {
+        const { ui, metadata } = getState()
+        const program = metadata[ui.program.id]
```

Comment:
We have selectors for `ui.program.id` an `ui.input.type`, we should probably use them.